app/components/post_list/post_list.tsx

```diff
@@ -172,6 +173,10 @@ const PostList = ({
             }
             await fetchPostThread(serverUrl, rootId, options);
         }
+        const removalPromises = posts.
+            filter((post) => post.type === 'system_ephemeral').
+            map((post) => removePost(serverUrl, post));
```

Comment:
As my other comment, this removes the posts one by one. I think we could use `deletePosts`. @enahum What do you think? The function directly executes sql on the database, but seems like it should work here.

Reply:
Appreciate the suggestion 🙏

I tried the following code:
```typescript
const ephemeralPostIDs = posts.
    filter((post) => post.type === 'system_ephemeral').
    map((post) => post.id);
await deletePosts(serverUrl, ephemeralPostIDs);
```
For some reason the messages disappear only after I back out and re-enter the channel

Reply:
It kind of makes sense :( Since it is affecting directly the database, probably is bypassing all the RXJS logic, so... no reactivity.

Since we don't expect to have a troublesome amount of ephemeral posts at any moment, let's leave it as deleting the post one by one. We can revisit this if we ever find it becoming a performance issue.

Reply:
I apologize, but it finally dawned on me that we are deleting **ALL** ephemeral posts on the current server.  Shouldn't we limit it to this channel or thread?

This was the suggestions [here](https://github.com/mattermost/mattermost/issues/28408):

* When we do the pull up and refresh channel, remove only the ephemeral messages on that channel.
* When we do the pull up and refresh on a thread, remove only the ephemeral messages on that thread.

I think we should add channel_id or original_post_id to limit by channel and limit by thread?

Reply:
Makes sense. I'll take a look 👍

Reply:
I apologized, I made an error in my request.  The posts passed into the component should already be content for the channel or thread.  And filtering it by channelId or rootId is redundant and not necessary.

I have reverted your changes, and went back to the last commit. Sorry about that @theoforger.  I am learning as well.